src/pip/_internal/index.py

```diff
@@ -266,10 +266,27 @@ class CandidateEvaluator(object):
     def __init__(
         self,
         valid_tags,          # type: List[Pep425Tag]
-        prefer_binary=False  # type: bool
+        prefer_binary=False,   # type: bool
+        py_version=None,       # type: Optional[str]
+        py_version_info=None,  # type: Optional[Tuple[int, ...]]
```

Comment:
Do we really need `py_version` when we have `py_version_info` ?
`py_version` should always be equal `'.'.join(map(str, py_version_info[:2]))`

Reply:
I kept them separate because the two arguments are used to check different things (the version info embedded in the filename, and the "Requires-Python" metadata), and in some places pip has logic to support checking one but not the other (e.g. `--ignore-requires-python`). But I can change it to be the same here until / unless we need the additional flexibility.